src/tdastro/sources/physical_model.py

```diff
@@ -116,6 +116,49 @@ def set_graph_positions(self, seen_nodes=None):
         for effect in self.effects:
             effect.set_graph_positions(seen_nodes=seen_nodes)
 
+    def _redshift_pre_effect(self, observer_frame_times, observer_frame_wavelengths, params):
+        """Calculate the rest frame times and wavelengths needed to give user the observer frame times
+        and wavelengths (given the redshift).
+
+        Parameters
+        ----------
+        observer_frame_times : numpy.ndarray
+            The times at which the observation is made.
+        observer_frame_wavelengths : numpy.ndarray
+            The wavelengths at which the observation is made (in Angstroms).
+        params : `dict`
+            A dictionary of the model's parameters.
+
+        Returns
+        -------
+        tuple of (numpy.ndarray, numpy.ndarray)
+            The rest frame times and wavelengths needed to generate the rest frame flux densities,
+            which will later be redshifted  back to observer frame flux densities at the observer frame
+            times and wavelengths.
+        """
+        observed_times_rel_to_t0 = observer_frame_times - params["t0"]
```

Comment:
What happens if t0 is not set? We have a few options. We could skip the computation, fail a check, or use a reasonable default. How about something like:
```
t0 = params.get("t0", 0.0)
observed_times_rel_to_t0 = observer_frame_times - t0
```
or if we change how the parameters are passed (using `t0` and `redshift` instead of `params`) we could just make it:
```
observed_times_rel_to_t0 = observer_frame_times - t0
```

Reply:
Now raising ValueError when redshift is set but t0 is not (checking this in PhysicalModel.evaluate)